torch/csrc/jit/import.cpp

```diff
@@ -263,8 +263,25 @@ script::Module load(
     std::unique_ptr<ReadAdapterInterface> rai,
     c10::optional<c10::Device> device,
     script::ExtraFilesMap& extra_files) {
+  // Verify that we're loading a zip archive and not a torch.save pickle archive
+  // (marked by the 0x80 0x02 bytes at the start)
```

Comment:
maybe add a comment:

zip files by spec can start with any data, so technically they might start with 0x80 0x02, but in practice zip files start with a file entry which begins with 0x04034b50. Furthermore, PyTorch will never produce zip files that do not start with the file entry, so it is relatively safe to perform this check.